### PR TITLE
Add demo user online and offline test

### PR DESCRIPTION
**Pull request by ShivamPokhriyal**
#2341

### CODE REVIEW

**Comment by dimagimon** [comment]
[32% (0.0%) vs master 32%](https://jenkins.dimagi.com/job/commcare-android-pr-job-builder/462090/)

**Comment by dimagimon** [comment]
[32% (0.0%) vs master 32%](https://jenkins.dimagi.com/job/commcare-android-pr-job-builder/462091/)

File: app/instrumentation-tests/src/org/commcare/androidTests/DemoUserOfflineTest.kt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        installApp(APP_NAME, CCZ_NAME, true)
+    }
+
+    @Test
+    fun testPracticeMode_online() {
+        testPracticeMode()
+    }
+
+    @Test
+    fun testPracticeMode_withUpdatedApp_online() {
```
Comment (shubham1g5) [inline app/instrumentation-tests/src/org/commcare/androidTests/DemoUserOfflineTest.kt:31]: this would need to be suppressed below Android 8 right ?

**Review by shubham1g5 — CHANGES_REQUESTED** [review]
nice work with abstraction here. one minor miss.

File: app/instrumentation-tests/src/org/commcare/androidTests/DemoUserOfflineTest.kt (hunk 1 above)
Comment (ShivamPokhriyal) [reply app/instrumentation-tests/src/org/commcare/androidTests/DemoUserOfflineTest.kt:31]: Thanks for catching this one. I totally missed it.

**Comment by dimagimon** [comment]
[32% (0.0%) vs master 32%](https://jenkins.dimagi.com/job/commcare-android-pr-job-builder/462093/)

**Review by shubham1g5 — APPROVED** [review]

**Pull request merged**